Dockerfile

```diff
@@ -62,20 +62,17 @@ RUN set -o errexit \
  && set -o pipefail \
  && apk update \
  && apk upgrade \
- # install required packages
  && apk add --no-cache graphviz font-dejavu font-noto-cjk tomcat-native jetty-runner \
  # create group and user for plantuml
  && addgroup -S -g 1000 plantuml \
  && adduser -S -h /opt/apache-tomcat -s /bin/bash -G plantuml -u 1000 plantuml
 
 #install tomcat
-COPY --from=tomcat /opt/apache-tomcat-${TOMCAT_VERSION} ${CATALINA_BASE}
+COPY --from=tomcat --chown=plantuml:plantuml /opt/apache-tomcat-${TOMCAT_VERSION} ${CATALINA_BASE}
+COPY --from=builder --chown=plantuml:plantuml /src/plantuml-server-${PLANTUML_VERSION}/target/plantuml.war ${CATALINA_BASE}/webapps/
+COPY --chown=plantuml:plantuml resources /
 
-RUN chown -R plantuml:plantuml ${CATALINA_BASE}
-
-COPY --from=builder /src/plantuml-server-${PLANTUML_VERSION}/target/plantuml.war ${CATALINA_BASE}/webapps/
-
-COPY resources /
+USER 1000
```

Comment:
This syntax makes it to run process with UID=plantuml/1000 and GID=0/root.

I think this is counterproductive to the original task in removing root permissions

>  Warning
>
> When the user doesn’t have a primary group then the image (or the next instructions) will be run with the root group.
> ...

Source: https://docs.docker.com/engine/reference/builder/#user

Reply:
Ah, I overread the primary group part which `plantuml` already has. Total win